central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -87,17 +87,48 @@ func (ds *datastoreImpl) UpsertScanConfiguration(ctx context.Context, scanConfig
 }
 
 // DeleteScanConfiguration deletes the scan configuration specified by id
-func (ds *datastoreImpl) DeleteScanConfiguration(ctx context.Context, id string) error {
+func (ds *datastoreImpl) DeleteScanConfiguration(ctx context.Context, id string) (string, []string, error) {
 	if ok, err := scanConfigurationsSAC.WriteAllowed(ctx); err != nil {
-		return err
+		return "", nil, err
 	} else if !ok {
-		return sac.ErrResourceAccessDenied
+		return "", nil, sac.ErrResourceAccessDenied
 	}
 
 	ds.keyedMutex.Lock(id)
 	defer ds.keyedMutex.Unlock(id)
 
-	return ds.storage.Delete(ctx, id)
+	// first check if the scan configuration exists
+	scanConfig, found, err := ds.GetScanConfiguration(ctx, id)
+	if err != nil || !found {
+		return "", nil, errors.Errorf("Unable to find scan configuration id %q", id)
+	}
+	scanConfigName := scanConfig.GetScanName()
+
+	// get scan status for the scan configuration
+	scanStatuses, err := ds.GetScanConfigClusterStatus(ctx, id)
+	if err != nil {
+		return "", nil, errors.Errorf("Unable to find scan status for scan configuration id %q", id)
+	}
+
+	clustersDeleted := []string{}
```

Comment:
Actually, why do we need to know the clusters. Can't we just broadcast that the scan name is deleted or would that affect something else?

Reply:
that's a good point, I didn't know we could do that, would that be
 ```
m.sensorConnMgr.BroadcastMessage(sensorMessage)
```
instead of 
```
m.sensorConnMgr.SendMessage(clusterID, sensorMessage)
```